Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -0,0 +1,479 @@
+#include "WarpXOpenPMD.H"
+#include "FieldIO.H"  // for getReversedVec
+
+WarpXOpenPMDPlot::WarpXOpenPMDPlot(bool oneFilePerTS,
+                       const std::string& openPMDFileType)
+  :m_Series(nullptr),
+   m_OneFilePerTS(oneFilePerTS),
+   m_OpenPMDFileType(openPMDFileType)
```

Comment:
... the assignment is then done via
```suggestion
   m_OpenPMDFileType(std::move(openPMDFileType))
```

That way, the constructor can accept lvalues and rvalues and will move rvalues in-place without unnecessary copies.

References:
- https://stackoverflow.com/questions/7592630/is-pass-by-value-a-reasonable-default-in-c11
- https://en.cppreference.com/w/cpp/language/copy_elision
- https://juanchopanzacpp.wordpress.com/2014/05/11/want-speed-dont-always-pass-by-value/